web3/pm.py

```diff
@@ -17,11 +24,98 @@
 # It is not automatically available on a web3 object.
 # To use the `PM` module, attach it to your web3 object
 # i.e. PM.attach(web3, 'pm')
+
+
 class PM(Module):
+
+    def set_registry(self, address=None):
+        if not address:
+            self.registry = Registry.deploy_new_instance(self.web3)
+        else:
+            self.registry = Registry(address, self.web3)
+
     def get_package_from_manifest(self, manifest):
         pkg = Package(manifest, self.web3)
         return pkg
 
     def get_package_from_uri(self, uri):
         pkg = Package.from_uri(uri, self.web3)
         return pkg
+
+    def release_package(self, name, version, uri):
+        if not self.registry:
+            raise AttributeError("Registry has not been set for web3.pm")
+        if self.web3.eth.defaultAccount != self.registry.owner:
+            # change error
+            raise AttributeError("Owner doesn't have right permissions to release pkg.")
+        self.registry.release(name, version, uri)
+
+    def get_release_data_from_registry(self, name, version):
+        if not self.registry:
+            raise AttributeError("Registry has not been set for web3.pm")
+        return self.registry.get_release_data(name, version) 
+
+    def get_package_from_registry(self, name, version):
+        if not self.registry:
+            raise AttributeError("Registry has not been set for web3.pm")
+        release_uri = self.registry.get_release_data(name, version)
+        return self.get_package_from_uri(release_uri)
+
+
+class Registry(Contract):
+    def __init__(self, address, w3):
+        # only works with v.py registry in ethpm/assets
+        # abi verification for compatibility?
```

Comment:
How about checking the `runtime_bytecode`.  Should be a known value right?

Reply:
Yup, but wouldn't that require anybody who wants to use this class to use the `registry.v.py` implementation I wrote? Not sure if that's ideal. Ideally, I'd like to couple this class to the ERC1319 standard rather than a particular implementation in a particular language.